src/components/Label/Label.tsx

```diff
@@ -1,8 +1,73 @@
+import { LABEL_LOCAL, LABEL_REMOTE } from "../NotesProvider/consts/labels";
+import { type IDecoratedNote, NoteSource } from "../NotesProvider/types/DecoratedNote";
+import type { IStorageNote } from "../NotesProvider/types/StorageNote";
 import "./Label.css";
 import { useMemo } from "react";
 
+export type ILabel = {
+  label: string;
+  isActive: boolean;
+  parent: ILabel | null;
+  children: ILabel[];
+  notes: IDecoratedNote[];
+};
+
+export function generateLabelTree(notes: IDecoratedNote[]): ILabel[] {
+  const local: ILabel = {
+    label: LABEL_LOCAL,
+    isActive: true,
+    parent: null,
+    children: [],
+    notes: [],
+  };
+  const remote: ILabel = {
+    label: LABEL_REMOTE,
+    isActive: true,
+    parent: null,
+    children: [],
+    notes: [],
+  };
+
+  function addToTree(labelPath: string[], note: IDecoratedNote, currentNode: ILabel) {
+    if (labelPath.length === 0) {
+      currentNode.notes.push(note);
+      return;
+    }
+
+    const [head, ...rest] = labelPath;
+    let childNode = currentNode.children.find((child) => child.label === head);
+    if (!childNode) {
+      childNode = {
+        label: head,
+        isActive: true,
+        parent: currentNode,
+        children: [],
+        notes: [],
+      };
+      currentNode.children.push(childNode);
+    }
+    addToTree(rest, note, childNode);
+  }
+
+  for (const note of notes) {
+    for (const label of note.original.labels) {
+      const parts = label.trim().split("/");
+      const root = note.source === NoteSource.Local ? local : remote;
+      if (parts.length === 1) {
+        if (parts[0] === LABEL_LOCAL || parts[0] === LABEL_REMOTE) {
+          root.notes.push(note);
+          continue;
+        }
+      }
+      addToTree(parts, note, root);
+    }
+  }
+
+  return [local, remote];
+}
+
 export function Label({ label, prefix = "" }: { label: string; prefix?: string }) {
-  const backgroundColor = useMemo(() => labelToColor(label), [label]);
+  const backgroundColor = useMemo(() => labelToColor(label.split("/").pop() || ""), [label]);
```

Comment:
```suggestion
  const backgroundColor = useMemo(() => labelToColor(label), [label]);
```

Background color should depend on the whole "path" (i.e. remote label `v0.4.5` should have different color than a local one with the same name)

Reply:
it makes labels in notes and labels on the tree displays differently (w/ different color)

Reply:
Hmm, we should have full path in `notes` then.

Reply:
full path displayed on tree?
eg.
local
local/a
local/a/b
local/a/c